packages/@aws-cdk/aws-lambda/lib/function-base.ts

```diff
@@ -283,6 +300,10 @@ export abstract class FunctionBase extends Resource implements IFunction, ec2.IC
     const action = permission.action ?? 'lambda:InvokeFunction';
     const scope = permission.scope ?? this;
 
+    if (['lambda:InvokeFunction', 'lambda:*'].includes(action) && !qualified) {
+      Annotations.of(this).addWarning('Lambda has changed their authorization strategy, which may affect resource permissions on unqualified arns. See https://github.com/aws/aws-cdk/issues/19273');
```

Comment:
I'm sure we will have thoughts on the warning message, so I'm kicking that conversation off here. Are we going to send a CLI notice about this change as well?

Reply:
How about: "AWS Lambda has changed their authorization strategy, which may cause client invocations of the lambda function <ARN> to fail with Access Denied errors. If you are using a lambda Version or Alias, make sure to call grantInvoke or addPermission on the Version or Alias, and not the underlying Function. See <github>."

Is there a way to only give this warning if their app includes a  Version or Alias somewhere of this unqualified Function?

(Again, put this in a separate PR please)

Reply:
And yes, I think we are planning to do a CLI notice. We should work with the Lambda team to craft the right verbiage for that.